src/tango/ip/fd_netlink.c

```diff
@@ -111,6 +111,34 @@ fd_nda_type_to_label( uint nda_type ) {
 }
 
 
+FD_TLS fd_nl_t fd_nl_gbl = {0};
+
+
+fd_nl_t *
+fd_nl_get( void ) {
+  if( FD_UNLIKELY( fd_nl_gbl.init == 0 ) ) {
+    if( fd_nl_init( &fd_nl_gbl, 1 ) ) {
+      FD_LOG_ERR(( "Unable to initialize netlink" ));
+    }
+  }
+
+  return &fd_nl_gbl;
+}
+
+
+/* change the fd of the socket to the specified value */
+int
+fd_nl_set_fd( int fd ) {
```

Comment:
This reads a little strange to me. We don't need to change the fd, you can just use the one that gets assigned to it and pass that to allow_fds

Reply:
I changed the code to do exactly that. I must have dropped that commit somewhere. Will fix

Reply:
it was - it needed some cleanup